PR: Remove some redundant icons

oscargus opened the pull request:
<!--- Make sure to read the Contributing Guidelines:                   --->
<!--- https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md --->
<!--- and follow PEP 8, PEP 257 and Spyder's code style:               --->
<!--- https://github.com/spyder-ide/spyder/wiki/Dev:-Coding-Style      --->

## Description of Changes
Removed some redundant icon descriptions (and some unused imports).

Not sure if there are files that can be removed as a consequence, but I do not think so as all removed fa-icons are just duplicate lines and the spyder-icons are from a font file anyway.

<!--- Explain what you've done and why --->






### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

<!--- TYPE YOUR USER/NAME AFTER THE FOLLOWING: --->
I certify the above statement is true and correct: Oscar Gustafsson/oscargus

<!--- Thanks for your help making Spyder better for everyone! --->

ccordoba12 commented on spyder/utils/icon_manager.py:
```diff
@@ -114,7 +112,6 @@
     'bold':                    [('fa.bold',), {'color': MAIN_FG_COLOR}],
     'italic':                  [('fa.italic',), {'color': MAIN_FG_COLOR}],
     'genprefs':                [('fa.cogs',), {'color': MAIN_FG_COLOR}],
-    'exit':                    [('fa.power-off',), {'color': MAIN_FG_COLOR}],
```
This icon is still in use

https://github.com/spyder-ide/spyder/blob/8ca177c7eb9bba6c0e1978c221fa4a65a2ea0b8c/spyder/app/mainwindow.py#L927-L931

Please carefully review if the other icons you're trying to remove in this PR are still in use or not.

oscargus replied:
It is redefined on line 205 (formerly 210).

I was maybe not that clear what I actually did. Everything I removed is because of duplicate keys. I have not check if there are actually unused icons here, only duplicate definitions. (I have kept the one with a higher line number as that should be the one kept when running the original code, unless they were identical definitions, then I kept the one which seems to be "best positioned in the code".)

ccordoba12 replied:
Ok, sorry for that. Duplicates probably start to appear thanks to merges or rebases.

oscargus commented:
Btw, here is the list of icons that are currently not used explicitly. Some are clearly for different file formats, some for "Spyder Tour" and some are for plugins, so one cannot remove any without careful considerations. However, an icon like 
history24 is not used and doesn't use the file spyder/images/console/history24.png even if it was used:
`    'history24':               [('fa.history',), {'color': MAIN_FG_COLOR}],`

The point of the PR is simply to have less dead code.

```
'ArchiveFileIcon',
 'AudioFileIcon',
 'CFileIcon',
 'CodeFileIcon',
 'CppFileIcon',
 'CsharpFileIcon',
 'DollarFileIcon',
 'DriveHDIcon',
 'ExcelFileIcon',
 'ExclamationFileIcon',
 'GridFileIcon',
 'ImageFileIcon',
 'JavaFileIcon',
 'JavascriptFileIcon',
 'JsonFileIcon',
 'MarkdownFileIcon',
 'PDFIcon',
 'PowerpointFileIcon',
 'PowershellFileIcon',
 'PythonFileIcon',
 'RFileIcon',
 'SwiftFileIcon',
 'VideoFileIcon',
 'WindowsFileIcon',
 'WordFileIcon',
 'arrow-continue',
 'arrow-step-in',
 'arrow-step-out',
 'arrow-step-over',
 'attribute',
 'auto_reload',
 'cmdprompt',
 'cmdprompt_t',
 'condapackages',
 'console',
 'down',
 'drag-horizontal',
 'edit24',
 'filesaveas2',
 'folding.arrow_down_off',
 'folding.arrow_down_on',
 'folding.arrow_right_off',
 'folding.arrow_right_on',
 'font',
 'history24',
 'ipython_console_t',
 'keyword',
 'kill',
 'module',
 'profiler',
 'project_expanded',
 'pylint',
 'pythonpath_mgr',
 'reload',
 'set_workdir',
 'spyder.autopep8',
 'spyder.example',
 'spyder.line_profiler',
 'spyder.memory_profiler',
 'terminated',
 'text',
 'tour.close',
 'tour.end',
 'tour.home',
 'tour.next',
 'tour.previous'
```

ccordoba12 commented:
> Btw, here is the list of icons that are currently not used explicitly

Ok, thanks for looking at that. If you're completely sure that these icons are not used, then please remove them, along with their associated pngs (if any).

> The point of the PR is simply to have less dead code.

Ok, no problem with that :+1:

goanpeca commented:
Hi @oscargus just checking if this PR needs additional work, or if it is ready for final review.

Thanks!

oscargus commented:
I may just as well have a more careful look at the list of possibly dead icons above and see if I can remove some of them as well.

oscargus commented:
I have identified the icons that were not called in the code (as far as I can tell) and removed the corresponding lines. I have also removed the corresponding Spyder 2 icons and removed them from the list in NOTICE.txt. As far as I can tell, I have not broken anything...

oscargus commented:
Btw, I have not changed the overlong lines in icon_manager.py. Seems like pep8speaks does not check this PR for some reason anyway and not sure if it would have triggered any adjacent line checks anyway.
